src/lib/CMakeLists.txt

```diff
@@ -673,6 +673,16 @@ if (${ENABLE_NUMA_SUPPORT})
     set(LIBRARIES ${LIBRARIES} ${NUMA_LIBRARY} hpinuma_msource_s)
 endif()
 
+set(ERASE_SEGMENT_TYPES "" CACHE STRING "Erase iterators and accessors for these types (e.g., RunLength,FrameOfReference). Good for compile time, bad for run time (if these types end up being used)")
```

Comment:
Does checking for the valid segment types make sense? Otherwise I'll probably accidentally do something like `-DERASE_SEGMENT_TYPE="FrameOfReferrence"`

Should be just a call to `list(SEARCH <valid_types> my_type)` https://cmake.org/cmake/help/v3.12/command/list.html

Reply:
I had the same thought, but at the same time I didn’t feel good about adding redundant information yet again. #1660 shows that we will forget it.

In the end, I figured that (a) developers will only type this once in a while, and (b) nothing goes really wrong if it’s mistyped.

An alternative would be to pass the comma-separated string to the compiler and use compile-time string magic to split it there. That way, we could check for the existence of all type, but it feels like overkill.

Reply:
If you don't like it, that's fine. Just wanted to mention it.